CHANGELOG updates for IceSSL (#4020)
zeroc-ice/ice: CHANGELOG-3.8.md

- Add new `ice2slice` compiler that converts Slice files in the `.ice` format (used by Ice) into Slice files in the
`.slice` format (used by IceRPC).

- Removed the `slice2html` compiler, which was previously used to convert Slice documentation comments to HTML. Doxygen
  should be used to generate Slice API documentation.

- Removed the `--impl` argument from the Slice compiler.

- Removed the `--ice` and `--underscore` Slice compiler options. The Slice parser now accepts identifiers with the Ice
  prefix and underscores by default.

## IceSSL Changes

The ssl transport is no longer a plug-in. It is now built into the main Ice library and always available.

### Integration with Platform SSL Engines

Ice 3.8 introduces new IceSSL configuration APIs that allow you to configure the ssl transport using platform-native
SSL engine APIs. This provides significantly greater flexibility for advanced use cases.

- The ssl transport can now be fully configured programmatically, without relying on IceSSL properties.

> - Separate configurations for outgoing and incoming SSL connections are supported.
> - Per-ObjectAdapter SSL configurations are now possible for fine-grained control.

Reviewer: ```suggestion
- Separate configurations for outgoing and incoming SSL connections are supported.
- Per-ObjectAdapter configuration is also possible. 
```

  - If you want to reuse a servant implementation in a derived servant class, you need to define or override the
    `dispatch` method in all your servant classes. See the Ice/filesystem demo for an example.

## DataStorm Changes

- The DataStorm publisher/subscriber framework has bee integrated into the Ice distribution, and is not longer a
  separate product.

- Added support for running DataStorm callbacks using a custom executor, the executor can be set using `customExecutor`
  Node's constructor parameter.

> - Add `DataStorm.Node.Name` property that allows configure the name used by a DataStorm node. The node names must

Reviewer: ```suggestion
- Added `DataStorm.Node.Name` property that allows configuring the name used by a DataStorm node. The node names must
```
